src/ExpressionEvaluator/CSharp/Test/ResultProvider/DynamicViewTests.cs

```diff
@@ -0,0 +1,145 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Dynamic;
+using System.Linq;
+using Microsoft.CodeAnalysis.ExpressionEvaluator;
+using Microsoft.VisualStudio.Debugger.Clr;
+using Microsoft.VisualStudio.Debugger.Evaluation;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests
+{
+    public class DynamicViewTests : CSharpResultProviderTestBase
+    {
+        private static readonly DkmClrRuntimeInstance s_runtime = new DkmClrRuntimeInstance(new[] { typeof(Microsoft.CSharp.RuntimeBinder.RuntimeBinderException).Assembly });
+
+        [Fact]
+        public void MultipleMembers()
+        {
+            var expression = "o";
+            dynamic o = new ExpandoObject();
+            o.Philosophers = new object[] { "Pythagoras", "Lucretius", "Zeno" };
+            o.WhatsForDinner = "Crab Cakes";
+            o.NumForks = 2;
+
+            var type = new DkmClrType(s_runtime, (TypeImpl)o.GetType());
+            var value = CreateDkmClrValue((object)o, type);
+            var result = FormatResult(expression, value);
+
+            Verify(result,
+                EvalResult(expression, "{System.Dynamic.ExpandoObject}", "System.Dynamic.ExpandoObject", expression, DkmEvaluationResultFlags.Expandable));
+            var dynamicView = GetChildren(result).Last();
+            Verify(dynamicView,
+                EvalResult(Resources.DynamicView, Resources.DynamicViewValueWarning, "", "o, dynamic", DkmEvaluationResultFlags.Expandable | DkmEvaluationResultFlags.ReadOnly));
+            Verify(GetChildren(dynamicView),
+                EvalResult("NumForks", "2", "System.Int32", "new Microsoft.CSharp.RuntimeBinder.DynamicMetaObjectProviderDebugView(o).Items[0]", DkmEvaluationResultFlags.ReadOnly),
+                EvalResult("Philosophers", "{object[3]}", "System.Object[]", "new Microsoft.CSharp.RuntimeBinder.DynamicMetaObjectProviderDebugView(o).Items[1]", DkmEvaluationResultFlags.ReadOnly),
+                EvalResult("WhatsForDinner", "\"Crab Cakes\"", "System.String", "new Microsoft.CSharp.RuntimeBinder.DynamicMetaObjectProviderDebugView(o).Items[2]", DkmEvaluationResultFlags.ReadOnly));
+        }
+
+        [Fact]
+        public void ExceptionTypeMember()
```

Comment:
This is intended to mimic that scenario.  I *believe* we cannot tell the difference, because the results have already been evaluated (and captured as a DynamicProperty object) by the time we see them.